lib/arm/cpu_features.c

```diff
@@ -167,6 +167,56 @@ static u32 query_arm_cpu_features(void)
 	}
 	return features;
 }
+
+#elif defined(_WIN32)
+
+/*
+ * https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions
+ * "The ARM64 version of Windows presupposes that it's running on an ARMv8 or
+ * later architecture at all times. Both floating-point and NEON support are
+ * presumed to be present in hardware."
+ * Some (but not all) other features are exposed in various Windows APIs.
+ * However, the cached values of the CP15 regs can also be read out of the
+ * registry.
+ */
+
+#include <Windows.h>
+#include <stdio.h>
+
+/* Note this reads cpu 0 only */
```

Comment:
Is that okay?  Some ARM SoCs are weird and have different features on the different CPUs.  I know that Linux, at least, only declares support for the CPUs' common subset of features.  I'd guess that Windows does the same.  Is it possible that this method of reading the attributes of CPU 0 specifically would circumvent that and cause problems?

Maybe you want to just leave out SHA3 and DOTPROD for now, given that they aren't yet properly supported by the Windows API, apparently?

Reply:
seems like it's the way to test it: https://github.com/microsoft/SymCrypt/blob/master/lib/cpuid.c